modules/core/src/main/java/org/apache/synapse/config/xml/CloneMediatorFactory.java

```diff
@@ -98,17 +114,68 @@ public Mediator createSpecificMediator(OMElement elem, Properties properties) {
         mediator.setSequential(!asynchronousExe);
         
         Iterator targetElements = elem.getChildrenWithName(TARGET_Q);
+        int noOfTargets = 0;
         while (targetElements.hasNext()) {
         	Target target = TargetFactory.createTarget((OMElement)targetElements.next(), properties);
         	target.setAsynchronous(asynchronousExe);
             mediator.addTarget(target);
+            noOfTargets++;
+        }
+
+        if (iterations != null && noOfTargets > 1) {
+            handleException("When iterations attribute is defined only one target is allowed.");
         }
 
         addAllCommentChildrenToList(elem, mediator.getCommentsList());
 
         return mediator;
     }
 
+    /**
+     * This is a utility method to resolve dynamic attribute
+     * @param attributeValue the attribute value
+     * @param elem the OMElement
+     * @return QName of the clone element in xml configuration
+     */
+    private Value resolveDynamicAttribute(String attributeValue, OMElement elem) {
+        try {
+            String valueExpression = attributeValue.substring(1, attributeValue.length() - 1);
+            if (valueExpression.startsWith("json-eval(")) {
+                new SynapseJsonPath(valueExpression.substring(10, valueExpression.length() - 1));
+            } else {
+                new SynapseXPath(valueExpression);
+            }
+        } catch (JaxenException e) {
+            handleException("Invalid expression for attribute 'name' : " + attributeValue);
+        }
+        // ValueFactory for creating dynamic Value
+        ValueFactory nameValueFactory = new ValueFactory();
+        // create dynamic Value based on OMElement
+        return nameValueFactory.createValue("iterations", elem);
```

Comment:
shall we use a const for "iterations", since used in many places

Reply:
This is used in this place only. However I have added a string constant in this class.